src/lexer/scanner.go

```diff
@@ -511,8 +511,18 @@ func (s *Scanner) Scan() Token {
 				continue
 			}
 
-			errorhandling.NewLexicalError(s.currentLineFile, s.currentColumnFile, string(s.lexemBuffer)+string(currChar))
+			if len(string(s.lexemBuffer)) == 0 {
+				errorhandling.NewLexicalError(s.currentLineFile, s.currentColumnFile, string(currChar))
+			}
```

Comment:
What is the point to trigger a new error if our lexem buffer is empty? Independently of the error class, we will not be able to give information about the lexem.

Reply:
There is a bug where is possible to have an empty lexembuffer (remember that we append new caracters at the end of the scanner) and when the message is shown, there is no lexem. Thus, this new conditionals solve the problem.

Reply:
Okay, there is any unit tests that covers this case that you mentioned?